File: Serialize/src/main/java/org/example/MatchmakingStats.java
package org.example;

import org.example.queueTypes.QueueType;

public class MatchmakingStats {

    private int totalCandidates;
    private QueueType queueType;

    public MatchmakingStats() {

    }

    public MatchmakingStats(QueueType queueType) {
        this.queueType = queueType;
    }

    public MatchmakingStats(int totalCandidates, QueueType queueType) {
        this.totalCandidates = totalCandidates;
        this.queueType = queueType;
    }

    public int getTotalCandidates() {
        return totalCandidates;
    }

    public void setTotalCandidates(int totalCandidates) {
        this.totalCandidates = totalCandidates;
    }

    public QueueType getQueueType() {
        return queueType;
    }

    public void setQueueType(QueueType queueType) {
        this.queueType = queueType;
    }

}

File: MatchmakingStats.py
class MatchmakingStats:
    def __init__(self, total_candidates=0, queue_type=None):
        self.total_candidates = total_candidates
        self.queue_type = queue_type

    def get_total_candidates(self):
        return self.total_candidates

    def set_total_candidates(self, total_candidates):
        self.total_candidates = total_candidates

    def get_queue_type(self):
        return self.queue_type

    def set_queue_type(self, queue_type):
        self.queue_type = queue_type